src/scanner.c

```diff
@@ -239,8 +239,11 @@ bool tree_sitter_rescript_external_scanner_scan(
       if (lexer->lookahead == 'n') {
         advance(lexer);
         if (lexer->lookahead == 'd') {
-          // Ignore new lines before `and` keyword (recursive definition)
-          in_multiline_statement = true;
+          advance(lexer);
+          if (is_whitespace(lexer->lookahead)) {
```

Comment:
I think it should rather check against valid identifier characters (a-z, A-Z, apostrophe, $) rather than whitespace, because `and ` (trailing space) is more or less the same as `and(` (trailing lparen) or `and/*comment*/` (a comment right after). What do you think?

Also, you consume an extra character I’m not sure it won’t break anything. For example, the case with a comment `and/*comment*/`

Would you add some tests?

Also, you consume

Reply:
> I think it should rather check against valid identifier characters (a-z, A-Z, apostrophe, $) rather than whitespace, because and  (trailing space) is more or less the same as and( (trailing lparen) or and/*comment*/ (a comment right after). What do you think?

I think we should scan for white space as the scanner is after a line break `\n`, so if an `and` following a white space is found then `in_multiline_statement` is true.

Note that `and` is a reserved word, so `and` followed by any character is a syntax error.

A problem found in `main`. The code below is parsed, but it should be an error.

```rescript
let a = 1 andb = 2
```

```
(source_file [0, 0] - [1, 0]
  (let_declaration [0, 0] - [0, 18]
    (let_binding [0, 4] - [0, 9]
      pattern: (value_identifier [0, 4] - [0, 5])
      body: (number [0, 8] - [0, 9]))
    (let_binding [0, 13] - [0, 18]
      pattern: (value_identifier [0, 13] - [0, 14])
      body: (number [0, 17] - [0, 18]))))
```